connectors/http/http-base/src/main/java/io/camunda/connector/http/base/client/apache/ProxyHandler.java

```diff
@@ -0,0 +1,118 @@
+    protocol = fallbackOnHttpForInvalidProtocol(uncheckedProtocol);
+    try {
+      if (isValidProxyFromSystemProperties(protocol)) {
+        setFromSystemProperties();
+      } else if (isValidProxyFromEnvVars(protocol)) {
+        setFromEnvVars();
+        if (doesTargetMatchNonProxy(protocol, requestUrl)) {
+          host = null;
+        }
+      }
+    } catch (NumberFormatException e) {
+      throw new ConnectorInputException(
+          "Invalid proxy port: " + (System.getProperty(protocol + ".proxyPort")), e);
+    }
+  }
+
+  private void setFromEnvVars() {
+    host = System.getenv("CONNECTOR_" + protocol.toUpperCase() + "_PROXY_HOST");
+    port = Integer.parseInt(System.getenv("CONNECTOR_" + protocol.toUpperCase() + "_PROXY_PORT"));
+    user = System.getenv().getOrDefault("CONNECTOR_" + protocol.toUpperCase() + "_PROXY_USER", "");
+    password =
+        System.getenv()
+            .getOrDefault("CONNECTOR_" + protocol.toUpperCase() + "_PROXY_PASSWORD", "")
+            .toCharArray();
+    nonProxyHosts = System.getenv("CONNECTOR_" + protocol.toUpperCase() + "_PROXY_NON_PROXY_HOSTS");
+    sourceIsSystemProperties = false;
+  }
+
+  private void setFromSystemProperties() {
+    host = System.getProperty(protocol + ".proxyHost");
+    port = Integer.parseInt(System.getProperty(protocol + ".proxyPort"));
+    user = System.getProperty(protocol + ".proxyUser");
+    password =
+        Optional.ofNullable(System.getProperty(protocol + ".proxyPassword"))
+            .map(String::toCharArray)
+            .orElseGet(() -> new char[0]);
+    sourceIsSystemProperties = true;
+  }
+
+  public CredentialsProvider getCredentialsProvider(String uncheckedProtocol) {
+    BasicCredentialsProvider provider = new BasicCredentialsProvider();
+    if (StringUtils.isNotBlank(host)
+        && StringUtils.isNotBlank(user)
+        && ArrayUtils.isNotEmpty(password)) {
+      provider.setCredentials(
+          new AuthScope(host, port), new UsernamePasswordCredentials(user, password));
+    }
+    return provider;
+  }
+
+  public HttpHost getProxyHost(String uncheckedProtocol, String requestUri) {
+    return host == null
+        ? null
+        : new HttpHost(fallbackOnHttpForInvalidProtocol(uncheckedProtocol), host, port);
+  }
+
+  private boolean doesTargetMatchNonProxy(String protocol, String requestUri) {
+    return (nonProxyHosts != null
+        && Arrays.stream(nonProxyHosts.split("\\|"))
+            .anyMatch(
+                nonProxyHost ->
+                    requestUri.matches(nonProxyHost.replace(".", "\\.").replace("*", ".*"))));
+  }
+
+  public HttpRoutePlanner getRoutePlanner(String uncheckedProtocol, HttpHost proxyHost) {
+    if (sourceIsSystemProperties) {
+      return new SystemDefaultRoutePlanner(
+          DefaultSchemePortResolver.INSTANCE, ProxySelector.getDefault());
+    } else if (proxyHost != null) {
+      return new DefaultProxyRoutePlanner(proxyHost);
+    }
+    return null;
```

Comment:
Would be great to add some debug logs here. Otherwise it might be intransparent which config was applied.

Reply:
Added debug logging.